pyomo/util/latex_printer.py

```diff
@@ -0,0 +1,1484 @@
+            ScalarParam: handle_param_node,
+            _ParamData: handle_param_node,
+            IndexedParam: handle_param_node,
+            NPV_Numeric_GetItemExpression: handle_npv_numericGetItemExpression_node,
+            IndexedBlock: handle_indexedBlock_node,
+            NPV_Structural_GetItemExpression: handle_npv_structuralGetItemExpression_node,
+            str: handle_str_node,
+            Numeric_GetAttrExpression: handle_numericGetAttrExpression_node,
+            NPV_SumExpression: handle_sumExpression_node,
+            NPV_DivisionExpression: handle_division_node,
+        }
+        if numpy_available:
+            self._operator_handles[np.float64] = handle_num_node
+
+    def exitNode(self, node, data):
+        try:
+            return self._operator_handles[node.__class__](self, node, *data)
+        except:
+            raise DeveloperError(
+                'Latex printer encountered an error when processing type %s, contact the developers'
+                % (node.__class__)
+            )
+
+
+def analyze_variable(vr):
+    domainMap = {
+        'Reals': '\\mathds{R}',
+        'PositiveReals': '\\mathds{R}_{> 0}',
+        'NonPositiveReals': '\\mathds{R}_{\\leq 0}',
+        'NegativeReals': '\\mathds{R}_{< 0}',
+        'NonNegativeReals': '\\mathds{R}_{\\geq 0}',
+        'Integers': '\\mathds{Z}',
+        'PositiveIntegers': '\\mathds{Z}_{> 0}',
+        'NonPositiveIntegers': '\\mathds{Z}_{\\leq 0}',
+        'NegativeIntegers': '\\mathds{Z}_{< 0}',
+        'NonNegativeIntegers': '\\mathds{Z}_{\\geq 0}',
+        'Boolean': '\\left\\{ 0 , 1 \\right \\}',
+        'Binary': '\\left\\{ 0 , 1 \\right \\}',
+        # 'Any': None,
+        # 'AnyWithNone': None,
+        'EmptySet': '\\varnothing',
+        'UnitInterval': '\\mathds{R}',
+        'PercentFraction': '\\mathds{R}',
+        # 'RealInterval' :        None    ,
+        # 'IntegerInterval' :     None    ,
+    }
+
+    domainName = vr.domain.name
+    varBounds = vr.bounds
+    lowerBoundValue = varBounds[0]
+    upperBoundValue = varBounds[1]
+
+    if domainName in ['Reals', 'Integers']:
+        if lowerBoundValue is not None:
+            lowerBound = str(lowerBoundValue) + ' \\leq '
+        else:
+            lowerBound = ''
+
+        if upperBoundValue is not None:
+            upperBound = ' \\leq ' + str(upperBoundValue)
+        else:
+            upperBound = ''
+
+    elif domainName in ['PositiveReals', 'PositiveIntegers']:
+        # if lowerBoundValue is not None:
+        if lowerBoundValue > 0:
+            lowerBound = str(lowerBoundValue) + ' \\leq '
+        else:
+            lowerBound = ' 0 < '
+        # else:
+        #     lowerBound = ' 0 < '
+
+        if upperBoundValue is not None:
+            if upperBoundValue <= 0:
+                raise ValueError(
+                    'Formulation is infeasible due to bounds on variable %s' % (vr.name)
+                )
```

Comment:
There actually is an `InfeasibleConstraintException` in `pyomo.common.errors` that might be the right thing here (and in place of the ValueErrors below also).